frontend/src/metabase/admin/performance/components/StrategyForm.tsx

```diff
@@ -196,18 +219,26 @@ const StrategyFormBody = ({
         <FormButtons
           targetId={targetId}
           shouldAllowInvalidation={shouldAllowInvalidation}
```

Comment:
I think the invalidation payload needs to be sensitive to its context. When I tried to invalidate on a dashboard, it failed every time because it was  trying to invalidate database 44 instead of dashboard 44

I wonder if in testing you only had 1 or 2 dashboards, so it seemed to work because you had 1 or 2 databases. This is why I like to run a dev environment with lots of my own data, etc.

![invalidateFail](https://github.com/metabase/metabase/assets/30528226/c579db4f-7ec9-4d2b-8809-51c9f243ae25)
![Screen Shot 2024-05-03 at 9 49 23 AM](https://github.com/metabase/metabase/assets/30528226/c45e188d-dbed-4553-be0a-bada9a96cd51)